docs/src/dev/provider/gremlin-semantics.asciidoc

```diff
@@ -524,6 +524,58 @@ fully demonstrative of Gremlin step semantics. It is also hard to simply read th
 step is meant to behave. This section discusses the semantics for individual steps to help users and providers
 understand implementation expectations.
 
+[[all-step]]
+=== all()
+
+*Description:* Filters array data from the Traversal Stream if all of the array's items match the supplied predicate.
+
+*Syntax:* `all(Predicate)`
+
+[width="100%",options="header"]
+|=========================================================
+|Start Step |Mid Step |Modulated |Domain |Range
+|N |Y |N |`List`/`array`/`Iterable`/`Iterator` |`List`/`array`/`Iterable`/`Iterator`
+|=========================================================
+
+*Arguments:*
+
+* `Predicate` - The predicate to use to test each value in the array data.
+
+*Modulation:*
+
+None
+
+*Considerations:*
+
+Each value will be tested using the supplied predicate with any ERROR state being reduced to FALSE. Empty lists always
+pass through and null/non-list traversers will be filtered out of the Traversal Stream.
+
+[[any-step]]
+=== any()
+
+*Description:* Filters array data from the Traversal Stream if any of the array's items match the supplied predicate.
+
+*Syntax:* `any(Predicate)`
+
+[width="100%",options="header"]
+|=========================================================
+|Start Step |Mid Step |Modulated |Domain |Range
+|N |Y |N |`List`/`array`/`Iterable`/`Iterator` |`List`/`array`/`Iterable`/`Iterator`
```

Comment:
Do these steps retain the initial `List`/`array`/`Iterable`/`Iterator` type or does everything get converted to `List`?

Reply:
These are filter steps so the traverser isn't modified. It will stay the same.